src/Slices/Settings/UI/Tabs/Configuration/Provider.tsx

```diff
@@ -53,33 +70,66 @@ export const Provider: React.FC<Props> = ({ settings: { settings, definition } }
     settings: settings,
     resetedValueName: "",
   });
+  const client = useQueryClient();
+  const [showUpdateBanner, setShowUpdateBanner] = useState(false);
   const [errorMessage, setErrorMessage] = useState("");
-  const { commandResolver } = useContext(DependencyContext);
-  const updateSetting = commandResolver.useGetTrigger<"UpdateEnvironmentSetting">({
-    kind: "UpdateEnvironmentSetting",
+
+  const updateSetting = useUpdateEnvironmentSetting({
+    onSuccess: () => {
+      client.refetchQueries();
+      document.dispatchEvent(new Event("settings-update"));
+
+      setShowUpdateBanner(true);
+      setTimeout(() => {
+        setShowUpdateBanner(false);
+      }, 2000);
+    },
+    onError: (error) => setErrorMessage(error.message),
   });
-  const resetSetting = commandResolver.useGetTrigger<"ResetEnvironmentSetting">({
-    kind: "ResetEnvironmentSetting",
+
+  const resetSetting = useResetEnvironmentSetting({
+    onError: (error) => setErrorMessage(error.message),
   });
+
   const handleReset = (id: string) => {
     dispatch({ type: "reset", payload: id });
+    if (errorMessage) {
```

Comment:
is the if-case really necessary?  i would have added it the same way you did for the onError, to keep it consistent